UnicornRender/video/source/Primitives.cpp

```diff
@@ -128,9 +147,143 @@ Mesh& Primitives::Sphere(Mesh& mesh, float radius, uint32_t rings, uint32_t sect
         }
     }
 
-    mesh.SetMeshData(vertices, indices);
+    mesh->SetMeshData(vertices, indices);
 
     return mesh;
 }
+
+namespace
+{
+
+Mesh* ProcessMesh(aiMesh* mesh, aiScene const* scene, std::string const& dir);
+
+void ProcessNode(aiNode* node, aiScene const* scene, std::string const& dir, std::list<Mesh*>& meshes)
+{
+    assert(nullptr != node);
+    assert(nullptr != scene);
+
+    for (unsigned int i = 0; i < node->mNumMeshes; i++)
+    {
+        aiMesh* mesh = scene->mMeshes[node->mMeshes[i]];
+        meshes.push_back(ProcessMesh(mesh, scene, dir));
+    }
+    for (unsigned int i = 0; i < node->mNumChildren; i++)
+    {
+        ProcessNode(node->mChildren[i], scene, dir, meshes);
+    }
+}
+
+std::vector<std::string> LoadMaterialTextures(aiMaterial* mat, aiTextureType type, std::string const& directory)
+{
+    assert(nullptr != mat);
+
+    std::vector<std::string> textures;
+    for (unsigned int i = 0; i < mat->GetTextureCount(type); ++i)
+    {
+        aiString str;
+        mat->GetTexture(type, i, &str);
+        std::string path = str.C_Str();
+        std::replace(path.begin(), path.end(), '\\', '/');
+        textures.emplace_back(directory + "/" + path);
+    }
+    return textures;
+}
+
+Mesh* ProcessMesh(aiMesh* mesh, aiScene const* scene, std::string const& dir)
```

Comment:
This function probably deserves a couple of comments along the way, or a few sub-routines, to make it more readable.